aapcs64/aapcs64.rst

```diff
@@ -1515,22 +1515,27 @@ When a non-bit-field member follows a bit-field it is placed at the lowest accep
     When laying out fundamental data types it is possible to consider them all to be bit-fields with a width equal to the container size. The rules in `Bit-fields no larger than their container`_ can then be applied to determine the precise address within a structure.
 
 
-Volatile bit-fields--preserving number and width of container accesses
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Volatile bit-fields -- preserving number and width of container accesses
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When a volatile bit-field is read, its container must be read exactly once using the access width appropriate to the type of the container.
+When a volatile bit-field is read, and its container does not overlap with any non-bit-field member or any zero length bit-field member, its container must be read exactly once using the access width appropriate to the type of the container.
 
-When a volatile bit-field is written, its container must be read exactly once and written exactly once using the access width appropriate to the type of the container. The two accesses are not atomic.
+When a volatile bit-field is written, and its container does not overlap with any non-bit-field member or any zero length bit-field member, its container must be read exactly once and written exactly once using the access width appropriate to the type of the container. The two accesses are not atomic.
+
+.. note::
+  This ABI does not place any restrictions on the access widths of bit-fields where the container overlaps with a non-bit-field member or where the container overlaps with any zero length bit-field placed between two other bit-fields. This is because the C/C++ memory model defines these as being separate memory locations, which can be accessed by two threads simultaneously. For this reason, compilers must be permitted to use a narrower memory access width (including splitting the access into multiple instructions) to avoid writing to a different memory location. For example, in :c:`struct S { int a:24; char b; };` a write to :sym:`a` must not also write to the location occupied by :sym:`b`, this requires at least two memory accesses in all current Arm architectures. In the same way, in :c:`struct S { int a:24; int:0; int b:8; };`, writes to :sym:`a` or :sym:`b` must not overwrite each other.
```

Comment:
```:c:`struct S { int a:24; char b; };` a write to :sym:`a` 
```
This markup syntax is not pure rst syntax. You can use ``code`` (I hope this doesn't format). See the surrounding markup code for examples of how to transform your additions.